packages/react-app/src/views/Whitelist.tsx

```diff
@@ -32,25 +33,29 @@ export default function Whitelist({ networkProvider, web3Modal }: Props) {
   }, [contracts, networkProvider, userAddress]);
 
   const loadWeb3Modal = useCallback(async () => {
-    const provider = await web3Modal.connect();
+    try {
+      const provider = await web3Modal.connect();
 
-    setInjectedProvider(new ethers.providers.Web3Provider(provider));
-
-    provider.on("chainChanged", (chainId: string) => {
-      DEBUG && console.log(`chain changed to ${chainId}! updating providers`);
       setInjectedProvider(new ethers.providers.Web3Provider(provider));
-    });
 
-    provider.on("accountsChanged", () => {
-      DEBUG && console.log(`account changed!`);
-      setInjectedProvider(new ethers.providers.Web3Provider(provider));
-    });
+      provider.on("chainChanged", (chainId: string) => {
```

Comment:
We extracted the wallet related codes to a hook in Parcel-0 project:
To give an idea you can check it here. 
https://github.com/citydaoproject/parcel-0-claim-portal/blob/95acff89f7f27e3b5d3c125317c467348758cfa2/hooks/useWallet.tsx